test/e2e/load.go

```diff
@@ -128,97 +123,97 @@ var _ = Describe("Load capacity", func() {
 func computeRCCounts(total int) (int, int, int) {
 	// Small RCs owns ~0.5 of total number of pods, medium and big RCs ~0.25 each.
 	// For example for 3000 pods (100 nodes, 30 pods per node) there are:
-	//  - 500 small RCs each 5 pods
+	//  - 300 small RCs each 5 pods
 	//  - 25 medium RCs each 30 pods
 	//  - 3 big RCs each 250 pods
 	bigRCCount := total / 4 / bigRCSize
-	mediumRCCount := (total - bigRCCount*bigRCSize) / 3 / mediumRCSize
-	smallRCCount := (total - bigRCCount*bigRCSize - mediumRCCount*mediumRCSize) / smallRCSize
+	mediumRCCount := total / 4 / mediumRCSize
+	smallRCCount := total / 2 / smallRCSize
 	return smallRCCount, mediumRCCount, bigRCCount
 }
 
-// The function every few second scales RC to a random size and with 0.1 probability deletes it.
-// Assumes that given RC exists.
-func playWithRC(c *client.Client, wg *sync.WaitGroup, ns, name string, size int) {
-	By(fmt.Sprintf("Playing with Replication Controller %v", name))
-	defer GinkgoRecover()
-	defer wg.Done()
-	// Wait some time to prevent from performing all operations at the same time.
-	time.Sleep(time.Duration(rand.Intn(60)) * time.Second)
-	rcExist := true
-	// Once every 1-2 minutes perform scale of RC.
-	for start := time.Now(); time.Since(start) < simulationTime; time.Sleep(time.Duration(60+rand.Intn(60)) * time.Second) {
-		if !rcExist {
-			config := RCConfig{Client: c,
-				Name:      name,
-				Namespace: ns,
-				Image:     image,
-				Replicas:  size,
-			}
-			expectNoError(RunRC(config), fmt.Sprintf("creating rc %s in namespace %s", name, ns))
-			rcExist = true
-		}
-		// Scale RC to a random size between 0.5x and 1.5x of the original size.
-		newSize := uint(rand.Intn(size+1) + size/2)
-		expectNoError(ScaleRC(c, ns, name, newSize), fmt.Sprintf("scaling rc %s in namespace %s", name, ns))
-		// List all pods within this RC.
-		_, err := c.Pods(ns).List(labels.SelectorFromSet(labels.Set(map[string]string{"name": name})), fields.Everything())
-		expectNoError(err, fmt.Sprintf("listing pods from rc %v in namespace %v", name, ns))
-		// With probability 0.1 remove this RC.
-		if rand.Intn(10) == 0 {
-			expectNoError(DeleteRC(c, ns, name), fmt.Sprintf("deleting rc %s in namespace %s", name, ns))
-			rcExist = false
+func generateRCConfigs(totalPods int, c *client.Client, ns string) []*RCConfig {
+	configs := make([]*RCConfig, 0)
+
+	smallRCCount, mediumRCCount, bigRCCount := computeRCCounts(totalPods)
+	configs = append(configs, generateRCConfigsForGroup(c, ns, smallRCGroupName, smallRCSize, smallRCCount)...)
+	configs = append(configs, generateRCConfigsForGroup(c, ns, mediumRCGroupName, mediumRCSize, mediumRCCount)...)
+	configs = append(configs, generateRCConfigsForGroup(c, ns, bigRCGroupName, bigRCSize, bigRCCount)...)
+
+	return configs
+}
+
+func generateRCConfigsForGroup(c *client.Client, ns, groupName string, size, count int) []*RCConfig {
+	configs := make([]*RCConfig, 0)
```

Comment:
nit - you know how many of them you're creating, so you can create the slice of correct size straight away, instead of appending in the loop.